python/src/aac/lang/language_context.py

```diff
@@ -0,0 +1,182 @@
+        init=False, default=Factory(dict), validator=validators.instance_of(dict)
+    )
+
+    def add_definition_to_context(self, definition: Definition):
+        """
+        Add the Definition to the list of definitions in the LanguageContext.
+
+        Args:
+            definition: The Definition to add to the context.
+        """
+        new_definition = copy.deepcopy(definition)
+
+        if new_definition.name not in self.definitions_name_dictionary:
+            self.definitions_name_dictionary[new_definition.name] = new_definition
+            self.definitions.append(new_definition)
+        else:
+            logging.debug(
+                f"Did not add definition '{new_definition.name}' to the context because one already exists with the same name."
+            )
+
+        if definition.is_extension():
+            self._apply_extension_to_context(definition)
+
+    def add_definitions_to_context(self, definitions: list[Definition]):
+        """
+        Add the list of Definitions to the list of definitions in the LanguageContext, any extensions are added last.
+
+        Args:
+            definitions: The list of Definitions to add to the context.
+        """
+
+        extension_definitions = get_definitions_by_root_key("ext", definitions)
+
+        # Avoiding the use of sorted() since we already deepcopy each definition as we add it to the context.
+        for definition in definitions:
+            if definition not in extension_definitions:
+                self.add_definition_to_context(definition)
+
+        for extension_definitions in extension_definitions:
+            self.add_definition_to_context(extension_definitions)
+
+    def get_root_keys(self) -> list[str]:
+        """
+        Get the list of root keys as defined in the LanguageContext.
+
+        Returns a list of strings indicating the current root keys in the active context. These
+        keys may differ from those provided by the core spec since the LanguageContext applies definitions
+        from active plugins and user files, which may extend the set of root keys.
+        See :py:func:`aac.spec.get_root_keys()` for the list of root keys provided by the unaltered core AaC DSL.
+
+        Returns:
+            A list of strings, one entry for each root key in the LanguageContext.
+        """
+        return [field.get("name") for field in self.get_root_fields()]
+
+    def get_root_fields(self) -> list[dict]:
+        """
+        Get the list of root fields as defined in the LanguageContext.
+
+        Returns a list of dictionaries that consist of the root fields in the active context. These
+        keys may differ from those provided by the core spec since the LanguageContext applies definitions
+        from active plugins and user files, which may extend the set of root keys.
+
+        Returns:
+            A list of dictionaries, one dictionary for each root field including name and type.
```

Comment:
```suggestion

        Returns:
            A list of dictionaries, one dictionary for each root field including name and type.
            
            These fields may differ from those provided by the core spec since the LanguageContext applies definitions
            from active plugins and user files, which may extend the set of root fields.
```